tools/LLVMOptimizer.cpp

```diff
@@ -0,0 +1,32 @@
+#include "LLVMOptimizer.h"
+
+using namespace llvm;
+
+
+LLVMOptimizer::LLVMOptimizer(std::string optArgs) {
+  this->optArgs = optArgs;
+  PB.registerModuleAnalyses(MAM);
+  PB.registerCGSCCAnalyses(CGAM);
+  PB.registerFunctionAnalyses(FAM);
+  PB.registerLoopAnalyses(LAM);
+  PB.crossRegisterProxies(LAM, FAM, CGAM, MAM);
+  
+  if(optArgs=="O2"){
+    MPM = PB.buildPerModuleDefaultPipeline(OptimizationLevel::O2);
```

Comment:
what about O1, O0, Oz, etc? Can't parsePassPipeline handle these?

Reply:
https://github.com/llvm/llvm-project/blob/main/llvm/tools/opt/opt.cpp#L794

LLVM opt uses if-else to take account teach scenarios. I can do the same thing here. What do you think?